### PR TITLE
Update to mbedtls 2.18.0

### LINKED ISSUE #42: Update to 2.18.0

### PR DESCRIPTION
**Pull request by jack-fortanix**
Closes #42

### CODE REVIEW

**Comment by jethrogb** [comment]
Hmm I think we should probably follow suit and split into two crates

**Comment by jack-fortanix** [comment]
I'm not sure if a split makes sense, because mbedtls still requires an embedded copy of mbed-crypto to build, and rexports mbed-crypto as part of the mbedtls API. So if we split into two crates, we end up with two copies of mbed-crypto in the resulting binary.

**Comment by jethrogb** [comment]
The embedded copy is used for more than just include files?

**Comment by jack-fortanix** [comment]
Yes it builds the version in `crypto/lib` to form `libmbedcrypto`. The crypto code in `mbedtls` itself is dead - if you check the `CMakeLists.txt` in `library` you see that `src_crypto` var is still set, but is just ignored now. No idea why they haven't removed that code entirely.

**Comment by jethrogb** [comment]
Ok let's keep it as a single crate for now, we can reevaluate later.

**Comment by jack-fortanix** [comment]
@jethrogb ping

**Comment by jethrogb** [comment]
I think you still need to bring over a bunch of changes we did internally.

**Comment by jack-fortanix** [comment]
Such as? Everything I see in the internal log is covered by #33 #41 #44

**Comment by jethrogb** [comment]
I had a bunch of comments on your internal PR, those don't seem to have been fixed here. e.g. `mbedtls-sys/build/config.rs`

**Comment by jack-fortanix** [comment]
Updated config.rs. No changes to headers.rs needed

File: mbedtls/src/bignum/mod.rs [hunk 1]
```diff
@@ -172,8 +172,9 @@ impl Mpi {
 
         let s = String::from_utf8(buf).expect("from_utf8 can't fail on radix-N data");
 
-        #[allow(deprecated)]
-        Ok(s.trim_right_matches(char::from(0)).to_owned())
+        // There is a null terminator plus (possibly) some garbage data. Remove it
+        let s : Vec<&str> = s.split(|c| c == char::from(0)).collect();
+        Ok(s[0].to_owned())
```
Comment (jethrogb) [inline mbedtls/src/bignum/mod.rs]: We fixed this internally as well

**Comment by jethrogb** [comment]
Can you update the nightly that's used?

**Pull request merged**